### PR TITLE
Add quickcheck tests to cover various ways of feeding the streaming h…

### PR DESCRIPTION
**Pull request by arthurprs**
…ashers

Continuation of #7

### CODE REVIEW

File: tests/quickcheck.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            h1.write(chunk);
+        }
+        let h1 = h1.finish();
+
+        let one_shot_result = xxhash_rust::xxh3::xxh3_64_with_seed(&xs, seed);
+
+        let sys_result = unsafe { sys::XXH3_64bits_withSeed(xs.as_ptr() as _, xs.len(), seed) };
+
+        // compare all, including to buffered and reference
+        let outcome = h0 == h1 && h0 == one_shot_result && h0 == sys_result;
```
Comment (DoumanAsh) [inline tests/quickcheck.rs:49]: Is it possible to make comparison more precise?
I'm not familiar with quick check, but the way it is done there is no way to know what went wrong exactly without breaking this test apart

Comment (arthurprs) [reply tests/quickcheck.rs]: Good point, I changed it to a standard assert_eq. On failures we'd get something like 
```
thread 'tests::xxh64_chunked_matches_buffered' panicked at '[quickcheck] TEST FAILED (runtime error). Arguments: (1, [], 1, 0)
Error: "assertion failed: `(left != right)`\n  left: `17241709254077376921`,\n right: `17241709254077376921`"
```

**Comment by DoumanAsh** [comment]
Thanks
Please do not format my code.... next time.
I'll revert these changes myself a bit later

**Review by DoumanAsh — APPROVED** [review]

**Comment by arthurprs** [comment]
It was unintentional, I must have ran rustfmt from the terminal. Let me revert that.

**Comment by DoumanAsh** [comment]
Ok thanks

**Comment by arthurprs** [comment]
@DoumanAsh I think it was xxh64.

**Comment by DoumanAsh** [comment]
ah that explains why they claim it is faster

**Pull request merged**